test/test_response.py

```diff
@@ -176,14 +175,11 @@ def test_io(self):
         assert resp.closed
 
         # Try closing with a base Response
-        try:
-            hlr = get_response()
-            resp2 = HTTPResponse(hlr.body, preload_content=False)
-            assert not resp2.closed
-            resp2.close()
-            assert resp2.closed
-        finally:
-            hlr.close()
+        hlr = get_response()
+        resp2 = HTTPResponse(hlr.body, preload_content=False)
+        assert not resp2.closed
+        resp2.close()
+        assert resp2.closed
```

Comment:
There's something funky about this whole `get_response` thing... `urllib3.base.Response` (and `urllib3.base`) seems to be something that Cory added in the v2 branch, with the idea that it would help with async support later (by having different subclasses for sync and async behavior). We're using a different approach, so I think we probably want to drop `urllib3.base` entirely...

Reply:
(to be clear though that's nothing to worry about in this pr, and I opened an issue so we don't forget)